source/device.hpp

```diff
@@ -39,20 +44,24 @@ class RoutingModule : public IRoutingDevice {
     virtual void add_inlink(std::shared_ptr<Link> link) final;
     virtual void update_routing_table(std::shared_ptr<IRoutingDevice> dest,
                                       std::shared_ptr<Link> link) final;
+    virtual std::vector<std::shared_ptr<IRoutingDevice>> get_neighbours()
+        const final;
+    virtual std::shared_ptr<Link> get_link_to_device(
+        std::shared_ptr<IRoutingDevice> device) const final;
 
     // returns next inlink and moves inlinks set iterator forward
     std::shared_ptr<Link> next_inlink() final;
 
-private:
+protected:
```

Comment:
There is no reason to use protected
```suggestion
private:
```